test/test_pebble.py

```diff
@@ -0,0 +1,512 @@
+            "kind": "autostart",
+            "err": "SILLY",
+            "ready": True,
+            "ready-time": "2021-01-28T14:37:04.291517768+13:00",
+            "spawn-time": "2021-01-28T14:37:02.247202105+13:00",
+            "status": "Done",
+            "summary": 'Autostart service "svc"',
+            "tasks": [],
+        })
+        self.assertEqual(change.id, '70')
+        self.assertEqual(change.kind, 'autostart')
+        self.assertEqual(change.err, 'SILLY')
+        self.assertEqual(change.ready, True)
+        self.assertEqual(change.ready_time, datetime_nzdt(2021, 1, 28, 14, 37, 4, 291517))
+        self.assertEqual(change.spawn_time, datetime_nzdt(2021, 1, 28, 14, 37, 2, 247202))
+        self.assertEqual(change.status, 'Done')
+        self.assertEqual(change.summary, 'Autostart service "svc"')
+        self.assertEqual(change.tasks, [])
+
+
+class MockAPI(pebble.API):
+    def __init__(self):
+        self.requests = []
+        self.responses = []
+
+    def _request(self, method, path, query=None, body=None):
+        self.requests.append((method, path, query, body))
+        return self.responses.pop(0)
+
+
+class TestAPI(unittest.TestCase):
+    maxDiff = None
+
+    def setUp(self):
+        self.api = MockAPI()
+
+    def test_get_system_info(self):
+        self.api.responses.append({
+            "result": {
+                "version": "1.2.3",
+                "extra-field": "foo",
+            },
+            "status": "OK",
+            "status-code": 200,
+            "type": "sync"
+        })
+        info = self.api.get_system_info()
+        self.assertEqual(info.version, '1.2.3')
+        self.assertEqual(self.api.requests, [
+            ('GET', '/v1/system-info', None, None),
+        ])
+
+    def test_get_warnings(self):
+        empty = {
+            "result": [],
+            "status": "OK",
+            "status-code": 200,
+            "type": "sync"
+        }
+        self.api.responses.append(empty)
+        warnings = self.api.get_warnings()
+        self.assertEqual(warnings, [])
+
+        self.api.responses.append(empty)
+        warnings = self.api.get_warnings(select=pebble.WarningState.ALL)
+        self.assertEqual(warnings, [])
+
+        self.assertEqual(self.api.requests, [
+            ('GET', '/v1/warnings', {'select': 'pending'}, None),
+            ('GET', '/v1/warnings', {'select': 'all'}, None),
+        ])
+
+    def test_ack_warnings(self):
+        self.api.responses.append({
+            "result": 0,
+            "status": "OK",
+            "status-code": 200,
+            "type": "sync"
+        })
+        num = self.api.ack_warnings(datetime_nzdt(2021, 1, 28, 15, 11, 0))
+        self.assertEqual(num, 0)
+        self.assertEqual(self.api.requests, [
+            ('POST', '/v1/warnings', None, {
+                'action': 'okay',
+                'timestamp': '2021-01-28T15:11:00+13:00',
+            }),
+        ])
+
+    def get_mock_change_dict(self):
```

Comment:
Is this clearer as "build_mock_change_dict" ? I had thought it was actually doing a request of some sort (GET).
Or 
"canned_mock_change_dict" ?